manifests/init.pp

```diff
@@ -11,14 +11,26 @@
   $purge_apt_conf  = false,
 ){
 
-  # Validates that $http_proxy and $http_proxy_port are domain names and ports respectively.
-  if $http_proxy { validate_re($http_proxy, '^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$') }
-  if $http_proxy_port { validate_re($http_proxy_port, '^\d+$') }
+  # Validates that $http_proxy is a valid, usable domain name or IP address
+  if $http_proxy and size($http_proxy) > 3 {   # '.' is an RFC-valid domain name, but not usable in this context
+    if is_ipv4_address($http_proxy) or is_domain_name($http_proxy) {
+      $_http_proxy = $http_proxy
+    } elsif is_ipv6_address($http_proxy) { # IPv6 needs to be enclosed in []
+      $_http_proxy = enclose_ipv6($http_proxy)
```

Comment:
I noticed those, but both looked like the underlying tooling handled host & port independently.  For rpm, this is the case.   I'll fix it up for APT, since the underlying puppet module concatenates the host and port into a string in the erb template.